core/app/models/spree/store.rb

```diff
@@ -21,28 +22,37 @@ def self.current(domain = nil)
 
     def self.default
       Rails.cache.fetch("default_store") do
-        where(default: true).first_or_initialize
+        find_or_initialize_by(default: true)
       end
     end
 
+    def self.no_default?
+      where(default: true).count.zero?
+    end
```

Comment:
Please create a scope non_default -> where.not(default: true) and add method has_default?